src/unistd.rs

```diff
@@ -2466,12 +2480,12 @@ impl User {
               libc::size_t,
               *mut *mut libc::passwd) -> libc::c_int
     {
-        let bufsize = match sysconf(SysconfVar::GETPW_R_SIZE_MAX) {
+        let cbuf_max = match sysconf(SysconfVar::GETPW_R_SIZE_MAX) {
             Ok(Some(n)) => n as usize,
             Ok(None) | Err(_) => 1024 as usize,
         };
 
-        let mut cbuf = Vec::with_capacity(bufsize);
+        let mut cbuf = Vec::with_capacity(512);
```

Comment:
You were right the first time.  GETPW_R_SIZE_MAX is the suggested initial size of the buffer for password entries, not the maximum size (earlier versions of POSIX did specify the maximum size).